### ISSUE TITLE
Extract the udf client and language specific tests into separate repositories

### ISSUE THREAD

**Issue by tkilias** [exasol/script-languages-release#299]
## Background

- We plan to extract the flavor families (standard, python3-minimal, r4-minimal, python3-data-science, ...) and the udf client into separate repositories.
- However, the udf client and all flavor families need to run the same udfclient and language-specific tests to ensure that the udf client works properly in the environment of the flavor (language version, library version, library locations, ...)
- Besides that, each flavor family has its own tests which are flavor specific
- We want to share the udf client and language-specific tests with all flavor family and the udf client repositories
- However, we want to avoid using submodules, because they usually create more problems than they solve
- Coping the files into all repositories is not a solution, so we have two alternatives
  1. We could use a kind of installer which copies the test files from a remote repository into the current one, and we commit in the current repository
  2. We could use a  kind of package manager which allows pulling arbitrary packages or which clones the repositories somewhere and adds links to the current working copy
- We would like to specify per flavor which test files for directories the flavor needs. 
  - We would like to be able to specify local directories or files or 
  - A remote directory in a git repository at a specific git ref
- We already have the test config and we might be able to reuse this in this case. 
- The test runner could play here the package manager. Which means whenever the tests are executed it checks if all tests are already loaded and if not, it loads them or updates the already loaded repositories

**Comment by tomuben** [comment]
@tkilias maybe the tests could be also placed inside of a wheel package, and the UDF client and flavor repos have a dependencie to the wheel package.

**Comment by tkilias** [comment]
> @tkilias maybe the tests could be also placed inside of a wheel package, and the UDF client and flavor repos have a dependencie to the wheel package.

yes, something like this, I had also in mind, but maybe we should split it into multiple packages, one for basic tests which can be used UDFClient and flavors and then flavor specific ones
